rts/Sim/Weapons/WeaponDef.cpp

```diff
@@ -386,11 +382,11 @@ WeaponDef::WeaponDef(const LuaTable& wdTable, const std::string& name_, int id_)
 		dmgs.reserve(32);
 		dmgTable.GetPairs(dmgs);
 
-		for (const auto& pair: dmgs) {
-			const int type = damageArrayHandler.GetTypeFromName(pair.first);
-			if (type == 0)
+		for (const auto& [armorTypeName, damage] : dmgs) {
+			const int armorType = damageArrayHandler.GetTypeFromName(armorTypeName);
```

Comment:
Changed to structured binding. The `type` variable was also shadowing `WeaponDef::type` so I changed it to something better as well.